### PR TITLE
Add forwarding methods for scheduling for Output<> types

### PR DESCRIPTION
**Pull request by steven-johnson**
Uses the same technique as Buffer -> Runtime::Buffer. Fixed/removed all
of the unnecessary explicit-cast-to-Func that I could find.

### CODE REVIEW

**Comment by zvookin** [comment]
LGTM.

**Comment by zvookin** [comment]
(Not going to merge until Andrew/others get a chance to comment, but ok if someone else decides to merge.)

File: src/Generator.h [hunk 1]
```diff
@@ -1658,7 +1658,67 @@ class GeneratorInput : public Internal::GeneratorInputImplBase<T> {
...
+#define HALIDE_OUTPUT_FORWARD(method)                                       \
+    template<typename ...Args>                                              \
+    inline auto method(Args&&... args) ->                                   \
+        decltype(std::declval<Func>().method(std::forward<Args>(args)...)) {\
+        return get_func_ref().method(std::forward<Args>(args)...);          \
+    }
+
+    /** Forward schedule-related methods to the underlying Func. */
+    // @{
+    HALIDE_OUTPUT_FORWARD(align_bounds)
```
Comment (abadams) [inline src/Generator.h]: Does there need to be a distinct const version for the const ones? (assuming there are some)

**Comment by abadams** [comment]
LGTM

**Pull request merged**